app/models/user.rb

```diff
@@ -554,6 +554,37 @@ def reactions_to
     Reaction.for_user(self)
   end
 
+  def latest_activity
+    return unless registered == true
+
+    latest_activity = last_comment_at || last_article_at || latest_article_updated_at ||
+      last_reacted_at || profile_updated_at || last_moderation_notification
+
+    if latest_activity.strftime("%Y-%m-%d") == Time.zone.today.strftime("%Y-%m-%d")
+      "Today, #{latest_activity.strftime('%d %b')}"
+    elsif latest_activity.strftime("%Y-%m-%d") == Date.yesterday.strftime("%Y-%m-%d")
+      "Yesterday, #{latest_activity.strftime('%d %b')}"
+    else
+      latest_activity.strftime("%d %b, %Y")
+    end
+  end
+
+  def current_role
+    if admin?
+      "Admin"
+    elsif super_admin?
+      "Super Admin"
+    elsif tech_admin?
+      "Tech Admin"
+    elsif tag_moderator?
+      "Moderator"
+    elsif has_role?(:single_resource_admin, :any)
+      "Resource Admin"
+    else
+      "Regular Member"
+    end
+  end
```

Comment:
I think this should also move this entire method a the same view helper as per `latest_activity` (app/helpers/admin/users_helper.rb), since it only deals with a view layer concern. We can also add a test for the helper method.

```
  def cascading_high_level_roles(user)
    if user.super_admin?
      "Super Admin"
    elsif user.admin?
      "Admin"
    elsif user.trusted?
      "Moderator"
    end
  end
```

(P.S i couldn't come up with a good method name 😅)

There's a couple of product design concerns that I will bring up with Anuj about this:
1. Do we mean to show the trusted role or the tag moderator role. From what I can see on the dropdown in the member view they are different: 
<img width="419" alt="Screenshot 2022-04-05 at 14 04 16" src="https://user-images.githubusercontent.com/2786819/161749952-c5b860d0-f0a0-4120-a72c-da99f805401b.png">
One is a trusted base role and the other is a resource admin role for tag's. I need to do a bit of digging into the code to understand it better before chatting to Anuj, so I'm okay with keeping it like this for now and changing it in a subsequent PR when we understand it better. 

2. In addition, the role "Moderator" does not map to any role on the dropdown in the member view, which I suspect will confuse users. 

3. A person can be a resource admin but not necessarily trusted hence this role does not fit the cascading profile and feels confusing. For now I think we should omit it until I chat to Anuj further.

Reply:
> Do we mean to show the trusted role or the tag moderator role.
I believe that we mean to show the "Tag Moderator" role here, as the base roles (e.g. Trusted) are shown under the "Status" column for a user as opposed to in the form of a tag. [This](https://github.com/forem/forem/pull/17033#discussion_r837026294) comment might help a teeny bit.

> In addition, the role "Moderator" does not map to any role on the dropdown in the member view, which I suspect will confuse users.
Agreed! I will adjust this to read "Tag Moderator" for the time being.

> A person can be a resource admin but not necessarily trusted hence this role does not fit the cascading profile and feels confusing. For now I think we should omit it until I chat to Anuj further.
Okay, I will omit the "Resource Admin" tag for now!

Reply:
Coincidentally, we started a similar conversation on the product team today😀. 

This `Trusted` base role(status) is a bit confusing. Since it gives you access to some features, it should be part of a special role named `Moderator` and not a part of the status. 

The overall idea is to treat the base roles as status and remove them from this dropdown and allow them to configure near the status area on the member detail view top part. So we will figure this out on the product end.

I think it's good not to have a Moderator role tag since it's not part of any special role. Once we have a better structure of roles, we can add the moderator role tag. So let's not show any role tag for trusted status. We know it gives the ability to the moderate community but giving a tag makes it confusing.

So in cascading profile, we can have `Super admin` > `Admin` > `Resource admin`

Reply:
Perfect, thank you @anujbhavsar96! 

@juliannatetreault you can tackle this in this PR or the next, whatever makes sense to you.

Reply:
Will do!